README.rst
Python Client for Web Risk API
==============================

|ga| |pypi| |versions|

**Note:** Cloud Web Risk is not publicly available. You must be added to the allowlist in order to gain access. See `Setting up Web Risk`_ 
in the product documentation for a link to the sign-up form.

.. _Setting up Web Risk: https://cloud.google.com/web-risk/docs/quickstart

- `Client Library Documentation`_
- `Product Documentation`_

.. |ga| image:: https://img.shields.io/badge/support-ga-gold.svg
   :target: https://github.com/googleapis/google-cloud-python/blob/master/README.rst#ga-support
.. |pypi| image:: https://img.shields.io/pypi/v/google-cloud-webrisk.svg
   :target: https://pypi.org/project/google-cloud-webrisk/
.. |versions| image:: https://img.shields.io/pypi/pyversions/google-cloud-webrisk.svg
   :target: https://pypi.org/project/google-cloud-webrisk/

.. _Client Library Documentation: https://googleapis.dev/python/webrisk/latest
.. _Product Documentation:  https://cloud.google.com/web-risk

Quick Start
-----------

In order to use this library, you first need to go through the following steps:

1. `Select or create a Cloud Platform project.`_
2. `Enable billing for your project.`_
3. `Enable the Web Risk API.`_
4. `Setup Authentication.`_

.. _Select or create a Cloud Platform project.: https://console.cloud.google.com/project
.. _Enable billing for your project.: https://cloud.google.com/billing/docs/how-to/modify-project#enable_billing_for_a_project
.. _Enable the Web Risk API.:  https://cloud.google.com/web-risk
.. _Setup Authentication.: https://googleapis.dev/python/google-api-core/latest/auth.html

Supported Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^
Python >= 3.6

Deprecated Python Versions
^^^^^^^^^^^^^^^^^^^^^^^^^^
Python == 2.7.

The last version of this library compatible with Python 2.7 is google-cloud-webrisk==0.3.0 

Installation
~~~~~~~~~~~~

Install this library in a `virtualenv`_ using pip. `virtualenv`_ is a tool to
create isolated Python environments. The basic problem it addresses is one of
dependencies and versions, and indirectly permissions.

With `virtualenv`_, it's possible to install this library without needing system
install permissions, and without clashing with the installed system
dependencies.

.. _`virtualenv`: https://virtualenv.pypa.io/en/latest/


Mac/Linux
^^^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    source <your-env>/bin/activate
    <your-env>/bin/pip install google-cloud-webrisk


Windows
^^^^^^^

.. code-block:: console

    pip install virtualenv
    virtualenv <your-env>
    <your-env>\Scripts\activate
    <your-env>\Scripts\pip.exe install google-cloud-webrisk

Next Steps
~~~~~~~~~~

-  Read the `Client Library Documentation`_ for Web Risk API
   API to see other available methods on the client.
-  Read the `Web Risk API Product documentation`_ to learn
   more about the product and see How-to Guides.
-  View this `repository’s main README`_ to see the full list of Cloud
   APIs that we cover.

.. _Web Risk API Product documentation:  https://cloud.google.com/web-risk
.. _repository’s main README: https://github.com/googleapis/google-cloud-python/blob/master/README.rst


CHANGELOG.md
# Changelog

[PyPI History][1]

[1]: https://pypi.org/project/google-cloud-webrisk/#history
## [0.3.3](https://github.com/googleapis/python-webrisk/compare/v0.3.2...v0.3.3) (2022-06-09)


### Bug Fixes

* **deps:** require protobuf<4.0.0 on v0 branch ([#204](https://github.com/googleapis/python-webrisk/issues/204)) ([cb10f98](https://github.com/googleapis/python-webrisk/commit/cb10f9843a3f7527d1f6932111c5043a829b9a3e))

### [0.3.2](https://github.com/googleapis/python-webrisk/compare/v0.3.1...v0.3.2) (2022-03-26)


### Bug Fixes

* **deps:** require google-api-core >= 1.31.5, >= 2.3.2 on v0 release ([22d8847](https://github.com/googleapis/python-webrisk/commit/22d884771c9f0658a946a67b9357c977e3a98727))

### [0.3.1](https://github.com/googleapis/python-webrisk/compare/v0.3.0...v0.3.1) (2022-03-23)


### Bug Fixes

* **deps:** require google-api-core >= 1.31.5, >= 2.3.2 on v0 release ([#174](https://github.com/googleapis/python-webrisk/issues/174)) ([123445e](https://github.com/googleapis/python-webrisk/commit/123445e6dce3dbab7c9d7ba2cd8b76a85382d6d5))

## [0.3.0](https://www.github.com/googleapis/python-webrisk/compare/v0.2.0...v0.3.0) (2020-04-20)


### Features

* add v1 ([#11](https://www.github.com/googleapis/python-webrisk/issues/11)) ([8b1ddd5](https://www.github.com/googleapis/python-webrisk/commit/8b1ddd5beed899ab0d30dbd79282beb0b055c78d))

## 0.2.0

07-24-2019 17:55 PDT


### Implementation Changes
- Allow kwargs to be passed to create_channel (via synth). ([#8412](https://github.com/googleapis/google-cloud-python/pull/8412))

### New Features
- Add 'client_options' support, update list method docstrings (via synth). ([#8530](https://github.com/googleapis/google-cloud-python/pull/8530))

### Dependencies
- Bump minimum version for google-api-core to 1.14.0. ([#8709](https://github.com/googleapis/google-cloud-python/pull/8709))

### Documentation
- Link to googleapis.dev documentation in READMEs. ([#8705](https://github.com/googleapis/google-cloud-python/pull/8705))
- Add compatibility check badges to READMEs. ([#8288](https://github.com/googleapis/google-cloud-python/pull/8288))

### Internal / Testing Changes
- Add docs job to publish to googleapis.dev. ([#8464](https://github.com/googleapis/google-cloud-python/pull/8464))
- Declare encoding as utf-8 in pb2 files (via synth). ([#8372](https://github.com/googleapis/google-cloud-python/pull/8372))
- Add disclaimer to auto-generated template files. ([#8336](https://github.com/googleapis/google-cloud-python/pull/8336))
- Fix coverage in 'types.py' (via synth). ([#8171](https://github.com/googleapis/google-cloud-python/pull/8171))
- Add empty lines (via synth). ([#8078](https://github.com/googleapis/google-cloud-python/pull/8078))

## 0.1.0

04-16-2019 17:09 PDT


### New Features
- Initial release of Web Risk. ([#7651](https://github.com/googleapis/google-cloud-python/pull/7651))

### Documentation
- Add whitelist info to README. ([#7717](https://github.com/googleapis/google-cloud-python/pull/7717))


noxfile.py
# -*- coding: utf-8 -*-
#
# Copyright 2018 Google LLC
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#     https://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

# Generated by synthtool. DO NOT EDIT!

from __future__ import absolute_import
import os
import shutil

import nox


BLACK_VERSION = "black==19.3b0"
BLACK_PATHS = ["docs", "google", "tests", "noxfile.py", "setup.py"]

nox.options.error_on_missing_interpreters = True

if os.path.exists("samples"):
    BLACK_PATHS.append("samples")


@nox.session(python="3.8")
def lint(session):
    """Run linters.

    Returns a failure if the linters find linting errors or sufficiently
    serious code quality issues.
    """
    session.install("flake8", BLACK_VERSION, "click<8.1")
    session.run("black", "--check", *BLACK_PATHS)
    session.run("flake8", "google", "tests")


@nox.session(python="3.8")
def blacken(session):
    """Run black.

    Format code to uniform standard.

    """
    session.install(BLACK_VERSION, "click<8.1")
    session.run("black", *BLACK_PATHS)


@nox.session(python="3.8")
def lint_setup_py(session):
    """Verify that setup.py is valid (including RST check)."""
    session.install("docutils", "pygments")
    session.run("python", "setup.py", "check", "--restructuredtext", "--strict")


def default(session):
    # Install all test dependencies, then install this package in-place.
    session.install("mock", "pytest", "pytest-cov")
    session.install("-e", ".")

    # Run py.test against the unit tests.
    session.run(
        "py.test",
        "--quiet",
        "--cov=google.cloud.webrisk",
        "--cov=google.cloud",
        "--cov=tests.unit",
        "--cov-append",
        "--cov-config=.coveragerc",
        "--cov-report=",
        "--cov-fail-under=0",
        os.path.join("tests", "unit"),
        *session.posargs,
    )


@nox.session(python=["2.7", "3.5", "3.6", "3.7", "3.8"])
def unit(session):
    """Run the unit test suite."""
    default(session)


@nox.session(python=["2.7", "3.7"])
def system(session):
    """Run the system test suite."""
    system_test_path = os.path.join("tests", "system.py")
    system_test_folder_path = os.path.join("tests", "system")
    # Sanity check: Only run tests if the environment variable is set.
    if not os.environ.get("GOOGLE_APPLICATION_CREDENTIALS", ""):
        session.skip("Credentials must be set via environment variable")

    system_test_exists = os.path.exists(system_test_path)
    system_test_folder_exists = os.path.exists(system_test_folder_path)
    # Sanity check: only run tests if found.
    if not system_test_exists and not system_test_folder_exists:
        session.skip("System tests were not found")

    # Use pre-release gRPC for system tests.
    session.install("--pre", "grpcio")

    # Install all test dependencies, then install this package into the
    # virtualenv's dist-packages.
    session.install("mock", "pytest")

    session.install("-e", ".")

    # Run py.test against the system tests.
    if system_test_exists:
        session.run("py.test", "--quiet", system_test_path, *session.posargs)
    if system_test_folder_exists:
        session.run("py.test", "--quiet", system_test_folder_path, *session.posargs)


@nox.session(python="3.8")
def cover(session):
    """Run the final coverage report.

    This outputs the coverage report aggregating coverage from the unit
    test runs (not system test runs), and then erases coverage data.
    """
    session.install("coverage", "pytest-cov")
    session.run("coverage", "report", "--show-missing", "--fail-under=75")

    session.run("coverage", "erase")


@nox.session(python="3.8")
def docs(session):
    """Build the docs for this library."""

    session.install("-e", ".")
    session.install("sphinx<3.0.0", "alabaster", "recommonmark", "jinja2<3.1")

    shutil.rmtree(os.path.join("docs", "_build"), ignore_errors=True)
    session.run(
        "sphinx-build",
        "-W",  # warnings as errors
        "-T",  # show full traceback on exception
        "-N",  # no colors
        "-b",
        "html",
        "-d",
        os.path.join("docs", "_build", "doctrees", ""),
        os.path.join("docs", ""),
        os.path.join("docs", "_build", "html", ""),
    )
